### PR TITLE
Interface cleanups and documentation

### PR DESCRIPTION
**Pull request by g2p**
Another batch adding docs and removing things that shouldn't be exposed

### CODE REVIEW

File: src/wodan/wodan.mli [hunk 1]
```diff
@@ -15,15 +15,18 @@
...
+    {!Wodan.S} module supporting filesystem operations *)
 
 exception BadMagic
 
 exception BadVersion
 
 exception BadFlags
 
-exception BadCRC of Location.t
+exception BadCRC of int64
```
Comment (pascutto) [inline src/wodan/wodan.mli:35]: What is the rationale here? 
It seems to me like `Location.t` was a good abstraction?

Comment (g2p) [reply src/wodan/wodan.mli:35]: It's a good abstraction, but it's not exposed externally; the exception leaked a module that wasn't intended to be visible.

Comment (pascutto) [reply src/wodan/wodan.mli:35]: Ah indeed. Perhaps it deserves a docstring then, saying this is indeed a location (and not e.g. a CRC code).

Comment (g2p) [reply src/wodan/wodan.mli:35]: Added, along with docs for the other exceptions

File: src/wodan/wodan.mli [hunk 2]
```diff
@@ -15,31 +15,58 @@
...
+(** Raised when trying to open a superblock and it sets must-support flags that this version doesn't support *)
+
+exception BadCRC of int64
+(** Raised when a block doesn't have the expected CRC
 
-exception BadCRC of Location.t
+    The exception carries the logical offset of the block. *)
 
 exception BadParams
+(** Raised when {!SUPERBLOCK_PARAMS} passed to the Make functor don't match actual settings saved in the superblock *)
```
Comment (pascutto) [inline src/wodan/wodan.mli]: Those look like long lines; apparently we haven't enabled `parse-docstrings` in `.ocamlformat` yet; we should probably do that.

**Review by pascutto — DISMISSED** [review]

File: src/wodan/wodan.mli (hunk 2 above)
Comment (g2p) [reply src/wodan/wodan.mli]: Sure. Added

**Review by pascutto — APPROVED** [review]

**Pull request merged**